velox/common/base/AsyncSource.h

```diff
@@ -144,6 +145,39 @@ class AsyncSource {
     return timing_;
   }
 
+  /// This function assists the caller in ensuring that resources allocated in
+  /// AsyncSource are promptly released:
+  /// 1. Waits for the completion of the 'make_' function if it is executing in
+  /// the thread pool.
+  /// 2. Resets the 'make_' function if it has not started yet.
+  /// 3. Cleans up the 'item_' if 'make_' has completed, but the result 'item_'
+  /// has not been returned to the caller.
+  void close() {
+    if (closed_ || resultMoved_) {
+      return;
+    }
+    ContinueFuture wait;
+    {
+      std::lock_guard<std::mutex> l(mutex_);
+      if (making_ && !promise_) {
+        promise_ = std::make_unique<ContinuePromise>();
+        wait = promise_->getSemiFuture();
+      }
+      if (make_) {
```

Comment:
`else if (make_)`
`else if (item_)`

Reply:
explained as above, `else if (item_)`  can not be moved here.

Reply:
changed for the first else if, thanks.